src/sql-parser/src/ast/defs/statement.rs

```diff
@@ -1174,6 +1176,45 @@ impl<T: AstInfo> AstDisplay for AlterIndexStatement<T> {
 
 impl_display_t!(AlterIndexStatement);
 
+#[derive(Debug, Clone, PartialEq, Eq, Hash)]
+pub enum AlterSourceAction<T: AstInfo> {
+    SetOptions(Vec<CreateSourceOption<T>>),
+    ResetOptions(Vec<CreateSourceOptionName>),
+}
+
+#[derive(Debug, Clone, PartialEq, Eq, Hash)]
+pub struct AlterSourceStatement<T: AstInfo> {
+    pub source_name: UnresolvedObjectName,
+    pub if_exists: bool,
```

Comment:
I carried this over from `ALTER INDEX` ~mechanically, but I can get rid of it if it seems like a bad idea. Little effort either way.